### ISSUE TITLE
CLI scan ignores passed user / options parser seems broken

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#17414]
### Steps to reproduce
1. Create two users "user1" and "user2"
2. Login as "user1"
3. Create a folder "test"
4. Upload a file into "test"
5. On the CLI: `sudo -uwwwrun ./occ files:scan --path=test user1`
6. On the CLI: `sudo -uwwwrun ./occ files:scan user1`

### Expected result
Both commands can run

### Actual result
```
# sudo -uwwwrun ./occ files:scan --path=test user1
Unknown user test

# sudo -uwwwrun ./occ files:scan user1
Scanning file   /
Scanning folder /
Scanning file   /user1
Scanning file   /.locks
Scanning file   /.ocdata
Scanning file   /root
Scanning file   /index.html
Scanning file   /owncloud.log
Scanning folder /user1
Scanning file   /user1/thumbnails
Scanning file   /user1/files
Scanning file   /user1/cache
Scanning folder /user1/thumbnails
Scanning file   /user1/thumbnails/17
Scanning file   /user1/thumbnails/11
Scanning folder /user1/thumbnails/17
Scanning file   /user1/thumbnails/17/36-36.png
Scanning file   /user1/thumbnails/17/2048-1536-max.png
Scanning folder /user1/thumbnails/11
Scanning file   /user1/thumbnails/11/36-36.png
Scanning file   /user1/thumbnails/11/2048-2048-max.png
Scanning folder /user1/files
Scanning file   /user1/files/test
Scanning file   /user1/files/welcome.txt
Scanning folder /user1/files/test
Scanning file   /user1/files/test/P1040667.JPG
Scanning folder /user1/cache
Scanning folder /.locks
Scanning folder /root
Scanning file   /root/thumbnails
Scanning file   /root/files
Scanning folder /root/thumbnails
Scanning file   /root/thumbnails/3
Scanning folder /root/thumbnails/3
Scanning file   /root/thumbnails/3/36-36.png
Scanning file   /root/thumbnails/3/2048-2048-max.png
Scanning folder /root/files
Scanning file   /root/files/welcome.txt
Scanning file   /user1/
Scanning folder /user1/
Scanning file   /user1/thumbnails
Scanning file   /user1/files
Scanning file   /user1/cache
Scanning folder /user1/thumbnails
Scanning file   /user1/thumbnails/17
Scanning file   /user1/thumbnails/11
Scanning folder /user1/thumbnails/17
Scanning file   /user1/thumbnails/17/36-36.png
Scanning file   /user1/thumbnails/17/2048-1536-max.png
Scanning folder /user1/thumbnails/11
Scanning file   /user1/thumbnails/11/36-36.png
Scanning file   /user1/thumbnails/11/2048-2048-max.png
Scanning folder /user1/files
Scanning file   /user1/files/test
Scanning file   /user1/files/welcome.txt
Scanning folder /user1/files/test
Scanning file   /user1/files/test/P1040667.JPG
Scanning folder /user1/cache
```

It seems it has trouble parsing the command line arguments.
In the first case it thought that "test" was the user name, and in the second case it scanned everything instead of just the user's home.

@icewind1991 @DeepDiver1975 @nickvergessen 

### Versions
ownCloud 8.1 RC2 and master
php5-5.6.9-1.1.x86_64

**Comment by PVince81** [comment]
It doesn't seem related to sudo, when I login as the user I still get the error.

**Comment by PVince81** [comment]
Ok case 1 is solved, the command should be `occ files:scan --path="user1/test"`

But for the second case, it should scan only for "user1" and not all users.

**Comment by PVince81** [comment]
Also it seems that passing `--all` does not only scan the home folders but also a lot of files outside, like:
```
Scanning file   /owncloud.log
```

**Comment by nickvergessen** [comment]
it also scanned twice?

**Comment by oparoz** [comment]
Yes, second command is broken.
Scans everything and twice.

**Comment by PVince81** [comment]
@icewind1991 any idea why it would scan all users even though a user name was given ?

**Comment by MorrisJobke** [comment]
Was fixed for master. I will move this to 8.2 and add the 8.1.1 brown label once this is also fixed in 8.1.1

**Comment by DeepDiver1975** [comment]
will be fixed with https://github.com/owncloud/core/pull/17676

**Comment by caratsch** [comment]
@MorrisJobke Is it possible to backport this also to oC 8.0.9? We just found the same problem on our systems.

**Comment by MorrisJobke** [comment]
@caratsch I asked for the backport: https://github.com/owncloud/core/pull/17659#issuecomment-160554621

**Comment by MorrisJobke** [comment]
@caratsch The fix will be included in 8.0.10 ;)

**Comment by caratsch** [comment]
@MorrisJobke Thank you very much! :-)
